core/bin/zksync_tee_prover/src/main.rs

```diff
@@ -0,0 +1,356 @@
+        let request = TeeProofGenerationDataRequest {};
+        let response = self
+            .send_http_request::<TeeProofGenerationDataRequest, TeeProofGenerationDataResponse>(
+                request,
+                endpoint.clone(),
+            )
+            .await?;
+        match response {
+            TeeProofGenerationDataResponse::Success(tvi) => Ok(tvi),
+            TeeProofGenerationDataResponse::Error(err) => {
+                let err_msg = format!("Failed to get proof gen data: {:?}", err);
+                tracing::error!(err_msg);
+                Err(anyhow!(err_msg))
+            }
+        }
+    }
+
+    fn verify(&self, tvi: TeeVerifierInput) -> anyhow::Result<(Signature, L1BatchNumber, H256)> {
+        match tvi.verify() {
+            Err(e) => {
+                let err_msg = format!("L1 batch verification failed: {e}");
+                tracing::warn!(err_msg);
+                Err(anyhow!(err_msg))
+            }
+            Ok(verification_result) => {
+                let root_hash_bytes = verification_result.0.as_bytes();
+                let batch_number = verification_result.1;
+                let signature = self.signing_key.try_sign(root_hash_bytes).unwrap();
+                Ok((signature, batch_number, verification_result.0))
+            }
+        }
+    }
+
+    async fn submit_proof(
+        &self,
+        signature: Signature,
+        root_hash: H256,
+        endpoint: &Url,
+    ) -> anyhow::Result<()> {
+        let request = SubmitTeeProofRequest(Box::new(L1BatchTeeProofForL1 {
+            signature: signature.to_vec(),
+            pubkey: self.signing_key.verifying_key().to_sec1_bytes().into(),
+            proof: root_hash.as_bytes().into(),
+        }));
+        let response = self
+            .send_http_request::<SubmitTeeProofRequest, SubmitProofResponse>(
+                request,
+                endpoint.clone(),
+            )
+            .await?;
+        match response {
+            SubmitProofResponse::Success => {
+                tracing::info!("Proof was successfully submitted");
+                Ok(())
+            }
+            SubmitProofResponse::Error(error) => {
+                let err_msg = format!("Submission of the proof failed: {}", error);
+                tracing::error!(err_msg);
+                Err(anyhow!(err_msg))
+            }
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl Task for TeeProverTask {
+    fn id(&self) -> TaskId {
+        "tee_prover".into()
+    }
+
+    async fn run(self: Box<Self>, mut stop_receiver: StopReceiver) -> anyhow::Result<()> {
+        tracing::info!("Starting the task {}", self.id());
+
+        let attestation_endpoint = self.api_url.join("/tee/register_attestation")?;
+        let job_fetcher_endpoint = self.api_url.join("/tee/proof_inputs")?;
+        let submit_proof_endpoint = self.api_url.join("/tee/submit_proofs")?;
```

Comment:
You may implement a thin wrapper over `reqwest::Client` that will give you a nice Rust API to avoid passing manually created endpoints.

Reply:
Fixed: https://github.com/matter-labs/zksync-era/pull/2333/commits/16f11d56a46a9e4e2a176287ad314283ddf54b9c. ✅

I hope this is more or less how you envisioned it. If not, let me know.